### PR TITLE
internal/host/plugin: remove custom queries for secrets

### PR DESCRIPTION
**Pull request by vancluever**
This removes the custom queries for host catalog secrets, now that #1658
has been merged.

It also does a lot of cleanup surrounding the related oplog entries,
using db.WithOplog natively.

### CODE REVIEW

**Comment by vancluever** [comment]
@talanknight @jimlambrt one thing particular that I'm wondering is if the oplog changes are correct. This setup seems to match the pattern we use in static host catalogs, but with that said, we don't necessarily need to work with a separate table over there.

**Comment by jefferai** [comment]
You could take a look at the upsert query in `internal/host/plugin/job_set_sync.go`. In a related question -- can Create and Update now just be combined into Upsert?

File: internal/host/plugin/host_catalog_secret_test.go
Comment (talanknight) [inline internal/host/plugin/host_catalog_secret_test.go:159]: Instead of using SetColumnValues, which require you passing in the values into the map, consider this potentially more succinct Action.  Here and in `repository_host_catalog.go`
```suggestion
		Action: db.SetColumns([]string{"secret", "key_id"}),
```

File: internal/host/plugin/repository_host_catalog.go
Comment (talanknight) [inline internal/host/plugin/repository_host_catalog.go:332]: I think we still want to track this using tickets since we are updating 2 different tables (the catalog table and the catalog secret table) with the aggregate being the catalog.

So this would mean adding a db.NewOplogMsg to the delete and the create(onconflict) calls and writing the oplog entries with the messages like existed before.

**Review by talanknight — CHANGES_REQUESTED** [review]
Looks good for the most part.  @jimlambrt will need to step in if I gave bad guidance regarding the oplog, but I think I'm correct :).

**Comment by vancluever** [comment]
FYI going to hold off on this until the long-running branch is merged to `main` and then we can do this work against that.

File: internal/host/plugin/repository_host_catalog.go
Comment (vancluever) [reply internal/host/plugin/repository_host_catalog.go]: Fixed. Resolved this while I was resolving the merge conflict so that it still uses the tickets both in create and update.

File: internal/host/plugin/host_catalog_secret_test.go
Comment (vancluever) [reply internal/host/plugin/host_catalog_secret_test.go]: Done!

**Comment by vancluever** [comment]
@talanknight merge conflicts resolved and comments addressed. Can you give it another look?

**Comment by vancluever** [comment]
@talanknight @jefferai @jimlambrt looking to get this closed off. Can one of you give this a review? Thanks!

**Review by jefferai — APPROVED** [review]
Looks good to me, would like to get a review from @talanknight too though

File: internal/host/plugin/host_catalog_secret.go
Comment (talanknight) [inline internal/host/plugin/host_catalog_secret.go:91]: I don't think you need to define oplog() at all, do you?

**Review by talanknight — APPROVED** [review]
Looks good!

File: internal/host/plugin/host_catalog_secret.go
Comment (vancluever) [reply internal/host/plugin/host_catalog_secret.go:91]: Is there an idiomatic way to add oplog messages in write queries without it?

Static catalogs have a similar pattern ([reference](https://github.com/hashicorp/boundary/blob/0a83e21aa82004f72b0c52d38166c099594db2ce/internal/host/static/repository_host_catalog.go#L146-L173)), but it's just a separate function versus a method. This more follows the pattern we use in the plugin-based host catalogs (ie: [`oplog`](https://github.com/hashicorp/boundary/blob/0a83e21aa82004f72b0c52d38166c099594db2ce/internal/host/plugin/host_catalog.go#L122-L132) for `HostCatalog` itself).

Comment (vancluever) [reply internal/host/plugin/host_catalog_secret.go:91]: Oh right, you know what, I do remember now - I was looking at adding this into the credential maintenance job, but it needs to be removed here for sure. I'll fix that before merging.

**Comment by vancluever** [comment]
@talanknight @jefferai @jimlambrt still need final review for this after addressing @talanknight's comments.

**Review by talanknight — APPROVED** [review]

**Pull request merged**
